cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -292,118 +285,193 @@ fn storage_footprint_to_ledger_footprint(foot: &Footprint) -> Result<LedgerFootp
+        let new_expiration_ledger = current_ledger_seq + ledgers_to_expire;
+        if new_expiration_ledger <= expiration_ledger {
+            // The bump would be innefective
+            continue;
+        }
+        let size = (key.to_xdr()?.len() + unmodified_entry.to_xdr()?.len()) as u32;
+        let rent_change = LedgerEntryRentChange {
+            is_persistent: durability == Persistent,
+            old_size_bytes: size,
+            new_size_bytes: size,
+            old_expiration_ledger: expiration_ledger,
+            new_expiration_ledger,
+        };
+        rent_changes.push(rent_change);
+    }
     let read_bytes = calculate_unmodified_ledger_entry_bytes(
         footprint.read_only.as_vec(),
-        snapshot_source,
+        ledger_storage,
         false,
     )?;
-    let soroban_resources = hack_soroban_resources(SorobanResources {
+    let soroban_resources = SorobanResources {
         footprint,
         instructions: 0,
         read_bytes,
         write_bytes: 0,
-        extended_meta_data_size_bytes: 2 * read_bytes,
-    });
+        contract_events_size_bytes: 0,
+    };
     let transaction_resources = TransactionResources {
         instructions: 0,
         read_entries: u32::try_from(soroban_resources.footprint.read_only.as_vec().len())?,
         write_entries: 0,
         read_bytes: soroban_resources.read_bytes,
         write_bytes: 0,
-        metadata_size_bytes: soroban_resources.extended_meta_data_size_bytes,
         transaction_size_bytes: estimate_max_transaction_size_for_operation(
             &OperationBody::BumpFootprintExpiration(BumpFootprintExpirationOp {
                 ext: ExtensionPoint::V0,
                 ledgers_to_expire: ledgers_to_expire,
             }),
             &soroban_resources.footprint,
         )?,
+        contract_events_size_bytes: 0,
     };
-    let fee_configuration = get_fee_configuration(snapshot_source, bucket_list_size)?;
-    let (min_fee, ref_fee) =
-        compute_transaction_resource_fee_wrapper(&transaction_resources, &fee_configuration);
-    let transaction_data = SorobanTransactionData {
-        resources: soroban_resources,
-        refundable_fee: ref_fee,
-        ext: ExtensionPoint::V0,
+    finalize_transaction_data_and_min_fee(
+        &ledger_storage,
+        &transaction_resources,
+        soroban_resources,
+        &rent_changes,
+        current_ledger_seq,
+        bucket_list_size,
+    )
+}
+
+fn get_bumpable_entry_expiration_and_durability(
+    ledger_entry: &LedgerEntry,
+) -> Result<(u32, ContractDataDurability), Box<dyn error::Error>> {
+    // TODO: Are there any other acceptable entry types?
+    //       is the ContractCode entry type always considered persistent?
+    let res = match ledger_entry.data {
+        LedgerEntryData::ContractData(ref cd) => (cd.expiration_ledger_seq, cd.durability),
+        LedgerEntryData::ContractCode(ref cc) => (cc.expiration_ledger_seq, Persistent),
+        _ => {
```

Comment:
@dmkozh @sisuresh can you check whether this is correct? the durability is used to populate the rent changes.